### PR TITLE
Fix re-running failed jobs by committers without rebase

### PR DESCRIPTION
**Pull request by potiuk**
After #31451 there were cases where re-running failed builds by committers without rebasing the PR could have caused that the runs were scheduled on self-hosted runners, but the runner rejected the build because it came from non-trusted actor.

This PR brings back the calculation of runs-on based on PR user login rather than on the actor of the build.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @o-nikolas and others - that should fix the situation when you are "just" rerunning someone's job without rebasing and it fails immediately. The #31451 changed the decision to run on to use "actor"  which was incompatible with runner check.

This caused - for example - attempts to re-run the jobs fail in https://github.com/apache/airflow/pull/30637

**Review by vincbeck — APPROVED** [review]

File: dev/breeze/src/airflow_breeze/utils/selective_checks.py [hunk 1]
```diff
@@ -776,7 +778,21 @@ def runs_on(self) -> str:
         if self._github_repository == APACHE_AIRFLOW_GITHUB_REPOSITORY:
             if self._github_event in [GithubEvents.SCHEDULE, GithubEvents.PUSH]:
                 return RUNS_ON_SELF_HOSTED_RUNNER
-            if self._github_actor in COMMITTERS and USE_PUBLIC_RUNNERS_LABEL not in self._pr_labels:
+            actor = self._github_actor
+            if self._github_event in (GithubEvents.PULL_REQUEST, GithubEvents.PULL_REQUEST_TARGET):
+                try:
+                    actor = self._github_context_dict["event"]["pull_request"]["user"]["login"]
+                    get_console().print(
+                        f"[warning]The {actor} retrieved as actor from GITHUB_CONTEXT's"
```
Comment (jedcunningham) [inline dev/breeze/src/airflow_breeze/utils/selective_checks.py]:
```suggestion
                        f"[info]{actor} retrieved as actor from GITHUB_CONTEXT's"
```

2 things:

- Don't need "the" here, even if "The potiuk retrieved as actor" does sounds good :)
- Shouldn't this be an info, not warning?

File: dev/breeze/src/airflow_breeze/utils/selective_checks.py [hunk 2]
```diff
@@ -776,7 +778,21 @@ def runs_on(self) -> str:
...
+                        f" event.pull_request.user.login[/]"
+                    )
+                except Exception as e:
+                    get_console().print(f"[warning]Exception when reading user login: {e}[/]")
+                    get_console().print(
+                        f"[info]Could not find the actor from pull request, "
+                        f"falling back to the actor who triggered the PR {actor}:[/]"
```
Comment (jedcunningham) [inline dev/breeze/src/airflow_breeze/utils/selective_checks.py]:
```suggestion
                        f"falling back to the actor who triggered the PR: {actor}[/]"
```

Maybe?

**Review by jedcunningham — APPROVED** [review]

File: dev/breeze/src/airflow_breeze/utils/selective_checks.py (hunk 1 above)
Comment (potiuk) [reply dev/breeze/src/airflow_breeze/utils/selective_checks.py]: Warning is better. The colors to show are specifically designed to be easily distinguishable.

image


* Don't need "the" here, even if "The potiuk retrieved as actor" does sounds good :)

I wanted to start the sentence capitalized.

Comment (potiuk) [reply dev/breeze/src/airflow_breeze/utils/selective_checks.py]: I changed it to:

```
                        f"[warning]The actor: {actor} retrieved from GITHUB_CONTEXT's"
```

**Pull request merged**